tests/Functional/ContentEventRenderTest.php

```diff
@@ -183,42 +167,511 @@ public function testEventRegistrationDateDescription(): void {
+        'title' => 'Event instagram link',
+        'link_type' => 'instagram',
+      ],
+    ])->save();
+    $this->drupalGet($node->toUrl());
+
+    $social_links_assert = new SocialMediaLinksAssert();
+    $social_links_expected_values = [
+      'title' => 'Social media',
+      'links' => [
+        [
+          'service' => 'facebook',
+          'label' => 'Event facebook link',
+          'url' => 'http://www.example.com/event_facebook',
+        ], [
+          'service' => 'instagram',
+          'label' => 'Event instagram link',
+          'url' => 'http://www.example.com/event_instagram',
+        ],
+      ],
+    ];
+    $social_links_content = $this->assertSession()->elementExists('css', '#event-practical-information .ecl-social-media-follow');
+    $social_links_html = $social_links_content->getOuterHtml();
+    $social_links_assert->assertPattern($social_links_expected_values, $social_links_html);
+    $social_links_assert->assertVariant('horizontal', $social_links_html);
+
+    // Unpublished Venues and Contacts are not visible for the visitors.
+    $contact_entity_general->setUnpublished()->save();
+    $contact_entity_press->setUnpublished()->save();
+    $venue_entity->setUnpublished()->save();
+
+    $this->drupalGet($node->toUrl());
+    $this->assertSession()->elementNotExists('css', '#event-contacts');
+
+    $field_list_expected_values['items'][1] = [
+      'label' => 'When',
+      'body' => "Wednesday 29 January 2020, 01:00\n to Saturday 8 February 2020, 01:00",
+    ];
+    unset($field_list_expected_values['items'][0]);
+    $field_list_assert->assertPattern($field_list_expected_values, $practical_list_content->getOuterHtml());
+
+    $icons_text_expected_values = [
+      'items' => [
+        [
+          'icon' => 'file',
+          'text' => 'Financing',
+        ], [
+          'icon' => 'calendar',
+          'text' => "29 January 2020, 01:00\n to 8 February 2020, 01:00",
+        ], [
+          'icon' => 'livestreaming',
+          'text' => 'Live streaming available',
+        ],
+      ],
+    ];
+    $icons_text_assert->assertPattern($icons_text_expected_values, $details_list_content->getOuterHtml());
   }
 
   /**
-   * Sets site's timezone.
+   * Assets enabled register button.
    *
-   * @param string $timezone
-   *   Timezone.
+   * @param \Behat\Mink\Element\NodeElement $parent_element
+   *   Parent element.
    */
-  protected function setTimezone(string $timezone): void {
-    \Drupal::configFactory()->getEditable('system.date')
-      ->set('timezone.default', $timezone)
-      ->save();
+  protected function assertRegisterButtonEnabled(NodeElement $parent_element): void {
+    $rendered_button = $this->assertSession()->elementExists('css', 'a.ecl-u-mt-2xl.ecl-link.ecl-link--cta', $parent_element);
+    $this->assertEquals('Register here', $rendered_button->getText());
+    $this->assertEquals('http://www.example.com/registation', $rendered_button->getAttribute('href'));
```

Comment:
Better to never hardcode assertion values, even if the assertion method is used only once.